nav2_velocity_smoother/src/velocity_smoother.cpp

```diff
@@ -274,22 +274,29 @@ double VelocitySmoother::findEtaConstraint(
     v_component_min = decel / smoothing_frequency_;
   }
 
+  if (std::abs(v_cmd) < 1e-6) {
+    // This avoids generating an eta of inf, which would result in sudden accelerations
+    // of the other factors.
+    return 1.0;
+  }
+
   if (dv > v_component_max) {
-    return v_component_max / dv;
+    return ( v_component_max + v_curr ) / v_cmd;
   }
 
   if (dv < v_component_min) {
-    return v_component_min / dv;
+    return ( v_component_min + v_curr ) / v_cmd;
   }
-
-  return -1.0;
+  // no acceleration limit exceeded, so return "no change"
+  return 1.0;
```

Comment:
If there's no change, then -1 would be the right return because if `eta < 0` we skip updating `curr_eta`